src/components/Button/styles.ts

```diff
@@ -1,21 +1,58 @@
 import styled from "styled-components/native";
+import { ButtonSize, ButtonVariant } from ".";
+import { LinearGradient } from "expo-linear-gradient";
+import {
+  getBackgroundStyles,
+  getBorderStyles,
+  getShadowStyles,
+} from "./Colors";
+import { getSpacingStyles } from "./Spacing";
 
-export const ButtonContainer = styled.TouchableOpacity`
-  background-color: ${({ theme }) =>
-    theme.colors.components.button.primary.background.default};
-  padding: 15px 20px;
-  border-radius: 10px;
+export interface ButtonBaseProps {
+  size: ButtonSize;
+  isPressed: boolean;
+  variant: ButtonVariant;
+  disabled: boolean;
+}
+
+export const Container = styled.TouchableOpacity<ButtonBaseProps>`
```

Comment:
N deu de usar o Pressable?
```suggestion
export const Container = styled.Pressable<ButtonBaseProps>`
```